### PR TITLE
Badge: Add tests for the badge component

### LINKED ISSUE #277: Write tests for the Badge General Component.

### PR DESCRIPTION
**Pull request by ritikamotwani**
Fixes: #277 

Tests for the badge component.

### CODE REVIEW

File: src/General/Badge/Badge.test.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+it(`<Badge> should render a label with the text "6"`, () => {
+  const BadgeSnapshot = renderer.create(<Badge label={label} />).toJSON();
+  expect(BadgeSnapshot).toMatchSnapshot();
+});
+
+it('should display the correct text when rendered', () => {
+  const { getByText } = render(<Badge label={label} />);
+  const BadgeContent = getByText(label);
+  expect(BadgeContent).toBeTruthy();
```
Comment (tingtingdisco) [inline src/General/Badge/Badge.test.tsx]: `expect(BadgeContent).toBeInTheDocument()` or `expect(BadgeContent).toBeVisible()`  is more declarative, and `getByText` will throw error if no element is found, so I use `queryByXXX` more, which returns `null` when no element found.

File: src/General/Badge/Badge.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+});
+
+describe('when sup is:', () => {
+  it('true, should have a top property', () => {
+    const { getByRole } = render(<Badge label={label} sup />);
+    const badgeContainer = getByRole('presentation');
+    const { top } = getComputedStyle(badgeContainer);
```
Comment (tingtingdisco) [inline src/General/Badge/Badge.test.tsx]: you can assert
`expect(badgeContainer).toHaveStyle('top: -.6em');`

File: src/General/Badge/Badge.tsx [hunk 3]
```diff
@@ -13,6 +13,7 @@ const Badge: React.FunctionComponent<Props> = ({
     {!isNil(label) && (
       <BadgeContainer
         className={classNames('aries-badge', className)}
+        role="presentation"
```
Comment (tingtingdisco) [inline src/General/Badge/Badge.tsx:16]: why name the role `presentation`?

Comment (ritikamotwani) [reply src/General/Badge/Badge.tsx:16]: Because it had the same name in the rest of the components, so followed that :stuck_out_tongue:

**Comment by ritikamotwani** [comment]
Updated it.

**Review by tingtingdisco — APPROVED** [review]
LGTM

**Comment by tingtingdisco** [comment]
ah @ritikamotwani sorry, could you add a test case to validate passed className?

**Pull request merged**